Source/WebCore/page/DOMSelection.cpp

```diff
@@ -550,8 +579,17 @@ unsigned DOMSelection::shadowAdjustedOffset(const Position& position) const
     if (position.isNull())
         return 0;
 
-    if (frame()->settings().liveRangeSelectionEnabled())
-        return shadowAdjustedNode(position) ? position.computeOffsetInContainerNode() : 0;
+    if (frame()->settings().liveRangeSelectionEnabled()) {
+        auto node = position.containerNode();
+        if (!node)
+            return 0;
+        if (node->isInShadowTree()) {
+            if (RefPtr host = node->shadowHost())
```

Comment:
This isn't quite right. We need to find the outermost shadow host instead.
frame()->document()->ancestorNodeInThisScope is the right thing to use here as the non-live range version does.

Reply:
(Ditto, re: above)